src/porepy/numerics/ad/equation_system.py

```diff
@@ -168,22 +168,31 @@ def __init__(self, mdg: pp.MixedDimensionalGrid) -> None:
 
         """
 
-        self._variables: list[Variable] = list()
```

Comment:
Internal maps are now all based on the variable ID for quick access.
_variable is a general collection containing the variables as created chronologically.
Other maps are slightly optimized (after each variable creation) to reflect a grid-wise clustering,